### PR TITLE
remove explicit population of GHRepository.parent since github-api does this now

### LINKED ISSUE #158: Remove explicit population of GHRepository.parent since github-api now does this
#155 brings in version 1.112 of `github-api` which fixes the following issue:
https://github.com/hub4j/github-api/pull/821/files

This is the relevant code (check to see if there are other references): https://github.com/salesforce/dockerfile-image-update/blob/4a5ee1679fb16c43e22e7c7e544a72aa0f3dc408/dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/process/GitHubPullRequestSender.java#L50-L52

### PR DESCRIPTION
**Pull request by RaviTezu**
Fixes #158

### CODE REVIEW

**Comment by RaviTezu** [comment]
@justinharringa could you please take a look? Thank you.

**Comment by RaviTezu** [comment]
@justinharringa @afalko please review this PR when you are free. Thanks.

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/process/GitHubPullRequestSender.java [hunk 1]
```diff
@@ -53,19 +53,13 @@ public GitHubPullRequestSender(DockerfileGitHubUtil dockerfileGitHubUtil, Forkab
...
-                parent = dockerfileGitHubUtil.getRepo(parentRepoName);
-                ShouldForkResult shouldForkResult = forkableRepoValidator.shouldFork(parent, ghContent, gitForkBranch);
-                if (shouldForkResult.isForkable()) {
-                    contentsShouldFork++;
-                    // fork the parent if not already forked
-                    ensureForkedAndAddToListForProcessing(pathToDockerfilesInParentRepo, parent, parentRepoName, ghContent);
-                } else {
-                    log.warn("Skipping {} because {}", parentRepoName, shouldForkResult.getReason());
-                }
-            } catch (IOException exception) {
```
Comment (justinharringa) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/process/GitHubPullRequestSender.java:67]: Hey @RaviTezu ! Thanks for your contribution! 

I just have a couple of questions:
* Have you taken a look to see if there are any other iterations of this in the codebase?
* Are we sure that removing the try/catch is safe here? I'm wondering if that was also covering some error cases for the other invocations here.

Comment (RaviTezu) [reply dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/process/GitHubPullRequestSender.java:67]: Thanks for the reply @justinharringa 
- Yes, I looked for other instances if we are explicitly populating GHRepository.parent and didn't find any. 
- The try/catch block was not needed once the call `dockerfileGitHubUtil.getRepo` has been removed. It is the only call which can raise an IOException. So I removed it.

Comment (justinharringa) [reply dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/process/GitHubPullRequestSender.java:67]: Great! Thanks @RaviTezu! Did you happen to try this out to make sure that it was able to successfully fork a found repo?

Comment (RaviTezu) [reply dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/process/GitHubPullRequestSender.java:67]: @justinharringa I didn't try it. I ran the unittests and they passed.

**Comment by justinharringa** [comment]
I tested it out and it looks good!

After exporting `git_api_url` and `git_api_token`, I ran `java -jar dockerfile-image-update/target/dockerfile-image-update-1.0-SNAPSHOT.jar -o justinharringa parent someregistry.tld/omundoebao 42 image-to-tag-store`

Image to tag store: https://github.com/dockerfile-image-update/image-to-tag-store
PR: https://github.com/justinharringa/somecontainers/pull/1

**Comment by justinharringa** [comment]
Thanks again @RaviTezu

**Review by justinharringa — APPROVED** [review]

**Comment by RaviTezu** [comment]
Thank you @justinharringa

**Pull request merged**